### PR TITLE
Use the gpio::Pin enum in nRF52-based boards instead of raw numbers.

### PR DESCRIPTION
**Pull request by gendx**
### Pull Request Overview

This pull request leverages the `gpio::Pin` enum for the nRF52-based boards. This enum allows to use more meaningful names like `P1_04` instead of `36`.

### Testing Strategy

This pull request was tested with Travis-CI.

### TODO or Help Wanted

N/A

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make formatall`.

### CODE REVIEW

**Comment by bradjc** [comment]
Thoughts on if we just switch to using the Pin type rather than usize? I have a start here: https://github.com/tock/tock/commit/2391fdb633b53384538cde86f677d4cb4edeb0db. I can finish if this seems reasonable.

**Comment by alevy** [comment]
Hrm... @bradjc yes this seems cleaner to me. @gendx ?

**Comment by niklasad1** [comment]
@bradjc yes, please.

File: boards/acd52832/src/main.rs [hunk 1]
```diff
@@ -119,84 +118,49 @@ pub unsafe fn reset_handler() {
 
     // GPIOs
     let gpio_pins = static_init!(
-        [&'static dyn kernel::hil::gpio::InterruptValuePin; 14],
+        [&'static dyn kernel::hil::gpio::InterruptValuePin; 7],
```
Comment (gendx) [inline boards/acd52832/src/main.rs]: @bradjc is this refactoring expected?

**Comment by gendx** [comment]
I also updated the nrf52840_dongle code, which was merged in the meantime.

File: boards/acd52832/src/main.rs (hunk 1 above)
Comment (bradjc) [reply boards/acd52832/src/main.rs]: Yes...the ACD pins seems to be all wrong, I probably copied them and forgot to update.

**Comment by bradjc** [comment]
I also want to change Pinmux, but haven't finished that yet.

**Comment by bradjc** [comment]
Hmm...maybe the pinmux change won't happen right now. It is weird to me that `Pinmux` implements copy:

https://github.com/tock/tock/blob/77f2a4915d29e4b3b9849970bb51ce9e71d851aa/chips/nrf5x/src/pinmux.rs#L19-L21

which means that the same pin mux can be passed to multiple drivers, which weakens the utility of the pinmux type. I tried removing the derive, but it leads to many other changes which I couldn't immediately fix.

**Comment by gendx** [comment]
> Hmm...maybe the pinmux change won't happen right now. It is weird to me that `Pinmux` implements copy:
> 
> https://github.com/tock/tock/blob/77f2a4915d29e4b3b9849970bb51ce9e71d851aa/chips/nrf5x/src/pinmux.rs#L19-L21
> 
> which means that the same pin mux can be passed to multiple drivers, which weakens the utility of the pinmux type. I tried removing the derive, but it leads to many other changes which I couldn't immediately fix.

This would be a larger refactoring indeed.

Looks like this PR is ready to merge then :)

**Pull request merged**